test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,169 @@
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint32_t unpacked_size, unpacked_checksum;
+    uint64_t o, s, c;
+    wt_off_t unpacked_offset;
+    c = o = s = 0;
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &o) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &s) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &c) == 0);
+
+    // Adjust the unpacked values as the block manager also does this to avoid storing large
+    // offsets.
+    unpacked_offset = (s == 0) ? (wt_off_t)o : (wt_off_t)(o + 1) * block_allocsize;
+    unpacked_size = (s == 0) ? (uint32_t)s : (uint32_t)s * block_allocsize;
+    unpacked_checksum = (uint32_t)c;
+
+    if (pack_size != 0) {
+        CHECK(unpacked_offset == pack_offset);
+        CHECK(unpacked_size == pack_size);
+        CHECK(unpacked_checksum == pack_checksum);
+    } else {
+        CHECK(unpacked_offset == 0);
+        CHECK(unpacked_size == 0);
+        CHECK(unpacked_checksum == 0);
+    }
+}
+
+static void
+test_pack_and_unpack_addr_cookie(
+  WT_BLOCK *block, wt_off_t pack_offset, uint32_t pack_size, uint32_t pack_checksum)
+{
+    const uint8_t *begin;
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    uint32_t checksum, obj_id, size;
+    size_t addr_size;
+    wt_off_t offset;
+    pp = p;
+
+    begin = (const uint8_t *)pp;
+    REQUIRE(__wt_block_addr_pack(
+              block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    addr_size = WT_PTRDIFF(pp, begin);
+    unpack_addr_cookie_and_check(begin, block->allocsize, pack_offset, pack_size, pack_checksum);
+
+    // Test the block manager's unpack function.
+    REQUIRE(__wt_block_addr_unpack(
+              NULL, block, begin, addr_size, &obj_id, &offset, &size, &checksum) == 0);
+    if (size != 0) {
+        CHECK(offset == pack_offset);
+        CHECK(size == pack_size);
+        CHECK(checksum == pack_checksum);
+    } else {
+        CHECK(offset == 0);
+        CHECK(size == 0);
+        CHECK(checksum == 0);
+    }
+}
+
+static void
+test_pack_and_unpack_addr_cookie_manual(
+  WT_BLOCK *block, std::vector<int64_t> cookie_vals, std::vector<uint8_t> &expected_packed_vals)
+{
+    const uint8_t *begin;
+    std::vector<uint8_t> packed(24, 0);
+    uint8_t *p = packed.data();
+    begin = (const uint8_t *)p;
+    REQUIRE(__wt_block_addr_pack(block, &p, WT_TIERED_OBJECTID_NONE, (wt_off_t)cookie_vals[0],
+              (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]) == 0);
+    CHECK(packed[0] == expected_packed_vals[0]);
+    CHECK(packed[1] == expected_packed_vals[1]);
+    CHECK(packed[2] == expected_packed_vals[2]);
+
+    unpack_addr_cookie_and_check(begin, block->allocsize, (wt_off_t)cookie_vals[0],
+      (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]);
+}
+
+TEST_CASE("Block addr pack and unpack", "[block_addr]")
+{
+    WT_BLOCK b;
+    WT_BM bm, *bmp;
+
+    b.allocsize = 1;
+    bmp = &bm;
+    bmp->block = &b;
+
+    SECTION("Pack and unpack address cookie 1")
```

Comment:
would be clearer if we can have a description of what we are testing for each section. Maybe we can move the comment above each `test_pack_and_unpack_addr_cookie` into the section instead.